packages/python-google-compute-engine/google_compute_engine/instance_setup/instance_setup.py

```diff
@@ -140,6 +155,27 @@ def _GenerateSshKey(self, key_type, key_dest):
 
     file_utils.SetPermissions(key_dest, mode=0o600)
     file_utils.SetPermissions('%s.pub' % key_dest, mode=0o644)
+    with open('%s.pub' % key_dest, 'r') as pk:
+        key_data = pk.read()
+
+    key_values = key_data.split()
+    if len(key_values) < 2:
+        return
```

Comment:
when is this the case and is it OK to just ignore it..?

Reply:
In theory it shouldn't ever be the case, but if there is something weird in the file, or if it's empty or something, I just return None and ignore it. As for whether it's ok to ignore stuff, for the near future this is designed to fail in a way that doesn't affect current behavior, so if things don't work as expected we just want to continue.